pkg/k8s/api/v1alpha1/edgedevice_types.go

```diff
@@ -82,13 +82,72 @@ type SocketSetting struct {
 	NetworkType  *string `json:"networkType,omitempty"`
 }
 
+type SecurityMode string
+
+const (
+	// SecurityModeNoSec No security
+	SecurityModeNone SecurityMode = "None"
+	// SecurityModePSK Pre-Shared Key
+	SecurityModeDTLS SecurityMode = "DTLS"
+)
+
+type DTLSMode string
+
+const (
+	// DTLSModePSK Pre-Shared Key
+	DTLSModePSK DTLSMode = "PSK"
+	// DTLSModeRPK Raw Public Key
+	DTLSModeRPK DTLSMode = "RPK"
+	// DTLSModeX509 X.509
+	DTLSModeX509 DTLSMode = "X.509"
+)
+
+type LwM2MSettings struct {
+	EndpointName string `json:"endpointName,omitempty"`
```

Comment:
can this be empty?

Reply:
no this field is required, how about add a kubebuilder comment for it?